### PR TITLE
[Phase 3] Add option for button scroll instead of true infinite scroll (Full Feature) (GH-4069)

### PR DESCRIPTION
**Pull request by RadhepS**
Closes: GH-4069

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html [hunk 1]
```diff
@@ -38,21 +38,85 @@
...
+              >
+                <div class="row">
+                  <cx-product-grid-item
+                    *ngFor="let product of model?.products"
+                    [product]="product"
+                    class="col-12 col-sm-6 col-md-4"
+                  ></cx-product-grid-item>
+                </div>
+                <div
+                  style="display: flex; justify-content: center; padding-top: 25px"
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html]: you should put the styling in the styling sheet, and not in-line

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html [hunk 2]
```diff
@@ -38,21 +38,85 @@
...
+                  !isInfiniteScroll || isMaxProducts || isLastPage || isEmpty
+                "
+                (scrolled)="scrollPage(model.pagination.currentPage + 1)"
+              >
+                <cx-product-list-item
+                  *ngFor="let product of model?.products"
+                  [product]="product"
+                  class="cx-product-search-list"
+                ></cx-product-list-item>
+                <div style="display: flex; justify-content: center;">
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html]: you should put the styling in the styling sheet, and not in-line

**Comment by RadhepS** [comment]
Note: Move configuration to feature-toggle.ts that has not been merged

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.spec.ts [hunk 3]
```diff
@@ -156,4 +263,225 @@ describe('ProductListComponent', () => {
...
+
+        component.appendProducts = true;
+        component.infiniteScrollOperations(mockModel1Page2);
+
+        const totalLength =
+          mockModel1.products.length + mockModel1Page2.products.length;
+
+        expect(component.model.products.length).toEqual(totalLength);
+        expect(component.model.products).toContain(
+          jasmine.objectContaining(mockModel1Page2.products[0])
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.spec.ts]: Add for the other product to make sure

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.spec.ts [hunk 4]
```diff
@@ -156,4 +261,219 @@ describe('ProductListComponent', () => {
...
+
+        fixture.detectChanges();
+
+        expect(component.productLimit).toEqual(2);
+        expect(component.model.products.length).toEqual(2);
+        expect(component.isMaxProducts).toBeTruthy();
+        expect(component.isLastPage).toBeFalsy();
+
+        expect(
+          el.query(By.css('.btn-link.back-to-top')).nativeElement
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.spec.ts]: apply queryAll to get all elements and choose using innterText/Html

File: projects/storefrontstyles/scss/components/product/list/_product-list.scss [hunk 5]
```diff
@@ -50,4 +50,17 @@
...
   }
+
+  .cx-scroll-buttons {
+    flex: 1 1 auto;
+    text-align: center;
+    margin: 10px;
+    text-decoration: underline;
+  }
+
+  .show-more {
```
Comment (bgambocjaviniar) [inline projects/storefrontstyles/scss/components/product/list/_product-list.scss]: remove for the fix using queryAll

File: projects/storefrontstyles/scss/components/product/list/_product-list.scss [hunk 6]
```diff
@@ -50,4 +50,17 @@
...
+  }
+
+  .back-to-top {
```
Comment (bgambocjaviniar) [inline projects/storefrontstyles/scss/components/product/list/_product-list.scss]: same as above

File: yarn.lock [hunk 7]
```diff
@@ -6559,6 +6559,13 @@ ng-packagr@^5.2.0:
     terser "^3.16.1"
     update-notifier "^3.0.0"
 
+ngx-infinite-scroll@^8.0.0:
```
Comment (bgambocjaviniar) [inline yarn.lock]: Do we need lock file?

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html [hunk 8]
```diff
@@ -38,21 +38,85 @@
...
               </div>
             </ng-container>
 
             <ng-container *ngIf="viewMode === ViewModes.List">
-              <cx-product-list-item
-                *ngFor="let product of model?.products"
-                [product]="product"
-                class="cx-product-search-list"
-              ></cx-product-list-item>
+              <div
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list.component.html]: breaking change

**Pull request merged**